id/_internal/oidc/ambient.py

```diff
@@ -258,3 +261,31 @@ def detect_buildkite(audience: str) -> Optional[str]:
         )
 
     return process.stdout.strip()
+
+
+def detect_env_var(audience: str) -> Optional[str]:
+    """
+    Detect and return ambient OIDC credential from an environment variable.
+
+    This detection works with GitLab pipeline `id_tokens`.
+
+    The environment variable name is "<AUD>_ID_TOKEN" where <AUD> is the uppercased
+    audience argument where all characters outside of ASCII letters and digits are
+    replaced with "_". So audience "sigstore" uses environment variable
+    SIGSTORE_ID_TOKEN and audience "http://test.audience" uses environment variable
+    HTTP___TEST_AUDIENCE_ID_TOKEN.
+
+    Returns `None` if a matching environment variable was not found for the audience.
+    """
+    logger.debug("Env var: looking for OIDC credentials")
```

Comment:
Might be good to include the env var name we're looking for in this message as well, especially since we could be significantly transforming it.

Reply:
env var name actually does get printed in every case: but I'll make sure that remains so with the changes